Source/WebGPU/WebGPU/WebGPUSwift.mm

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2021-2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "WebGPU.h"
+
+#if ENABLE(WEBGPU_SWIFT)
+#include "WebGPUExt.h"
+#include "WebGPUSwiftInternal.h"
+unsigned WebGPUTest::CppRefCounted::getSwiftRefCount()
+{
+    return WebGPU::TestPlumbing::getRefCount(this);
+}
+
+unsigned WebGPUTest::CppRefCounted::getSwiftRefCountOptimizedOut()
+{
+    return WebGPU::TestPlumbing::getRefCountOptimizedOut(this);
+}
```

Comment:
I want to expose the swift functions out of WebGPU.framework as a C++ API so that I can call it in TestWebKitAPI. Do you know of another way to do it ? If I directly want to call TestPlumbing then I will have to enable Interop in the test binary also.. which I really did not want to do.

Reply:
I think we want to unit test interop. And that's why this PR.
I just did not want to turn on interop in TestWebKitAPI yet unless we really have a pressing need for it. Usually, enabling interop involves yet another series of `#ifndef __swift__` which just adds debt to cleanup later.
If you strongly feel that we should just enable interop in TestWebKitAPI now, then I can work on that.